src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryFileValidator.cs

```diff
@@ -0,0 +1,153 @@
+            if (connInfo != null)
+            {
+                connInfo.TryGetConnection(Connection.ConnectionType.Default, out dbConn);
+                SqlConnection conn = ReliableConnectionHelper.GetAsSqlConnection(dbConn);
+                ServerConnection serverConnection = new ServerConnection(conn);
+
+                bool isLocal = false;
+                if (string.Compare(GetMachineName(serverConnection.ServerInstance), Environment.MachineName, StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    isLocal = true;
+                }
+
+                foreach (string filePath in args.FilePaths)
+                {
+                    bool IsFolder = false;
+                    bool Existing = IsPathExisting(serverConnection, filePath, ref IsFolder);
+
+                    if (Existing)
+                    {
+                        if (IsFolder)
+                        {
+                            errorMessage = string.Format(SR.BackupPathIsFolderError, filePath);
+                            result = false;
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        // If the file path doesn't exist, check if the folder exists
+                        string folderPath = PathWrapper.GetDirectoryName(filePath);
+                        if (isLocal)
+                        {
+                            if (!string.IsNullOrEmpty(folderPath) && !Directory.Exists(folderPath))
+                            {
+                                errorMessage = string.Format(SR.InvalidBackupPathError, folderPath);
+                                result = false;
+                                break;
+                            }
+                        }
+                        else
+                        {
+                            bool isFolderOnRemote = true;
+                            bool existsOnRemote = IsPathExisting(serverConnection, folderPath, ref isFolderOnRemote);
+                            if (!existsOnRemote)
+                            {
+                                errorMessage = string.Format(SR.InvalidBackupPathError, folderPath);
+                                result = false;
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+            else
+            {
+                result = false;
+            }
+
+            if (!string.IsNullOrEmpty(errorMessage))
+            {
+                throw new FileBrowserException(errorMessage);
+            }
+
+            return result;
+        }
+
+        #region private methods
+
+        private static bool IsPathExisting(ServerConnection serverConnection, string path, ref bool IsFolder)
+        {
+            Enumerator en = null;
+            DataSet ds = new DataSet();
```

Comment:
DataSet is disposable, you should dispose it when you're done with it's data

Reply:
k